core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -119,21 +151,26 @@ public void start(WorkerNetAddress address) throws IOException {
             ImmutableMap.of(),
             ImmutableMap.of(),
             Configuration.getConfiguration(Scope.WORKER));
+        LOG.info("Worker registered with worker ID: {}", mWorkerId.get());
         break;
       } catch (IOException ioe) {
         if (!retry.attempt()) {
           throw ioe;
         }
+      } finally {
+        mBlockMasterClientPool.release(masterClient);
       }
     }
   }
 
   @Override
   public void stop() throws IOException {
+    super.stop();
   }
 
   @Override
   public void close() throws IOException {
+    super.close();
```

Comment:
Do you think we should call `close()` in reverse order?
That is to call super.close() at the end of this function?

Reply:
fixed in [b884648](https://github.com/Alluxio/alluxio/pull/16867/commits/b88464885c3dad37953f1155e0df6d6f918b0853)